src/java/org/apache/cassandra/io/util/TailOverridingRebufferer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.io.util;
+
+import java.nio.ByteBuffer;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Special rebufferer that replaces the tail of the file (from the specified cutoff point) with the given buffer.
+ *
+ * Instantiated once per RandomAccessReader, thread-unsafe.
+ * The instances reuse themselves as the BufferHolder to avoid having to return a new object for each rebuffer call.
+ * Only one BufferHolder can be active at a time. Calling {@link #rebuffer(long)} before the previously obtained
+ * buffer holder is released will throw {@link AssertionError}.
+ */
+@NotThreadSafe
+public class TailOverridingRebufferer extends WrappingRebufferer
```

Comment:
nit: Only used in tests rn?

Reply:
This is the case but it will be used in src when CEP-25 hits. While I'm happy to make this move if you feel it necessary, my preference would be to leave it here since 25 is soon to be put to review.